refactron/analysis/symbol_table.py

```
@@ -61,6 +61,23 @@ class SymbolTable:
     exports: Dict[str, Symbol] = field(default_factory=dict)
+    # Metadata for caching (Map: file_path -> mtime)
+    metadata: Dict[str, float] = field(default_factory=dict)
+
+    def remove_file(self, file_path: str) -> None:
+        """Remove all symbols associated with a specific file."""
+        if file_path not in self.symbols:
+            return
+            
+        for scope, names in self.symbols[file_path].items():
+            if scope == "global":
+                for name, symbol in list(names.items()):
```

Ruff (0.15.9)

[warning] Loop control variable `symbol` not used within loop body

Rename unused `symbol` to `_symbol`

(B007)